### DISCUSSION TITLE
Issue with test fixture re-use clashing on multiple test classes

### DISCUSSION THREAD

**Discussion by ghost** [xunit/xunit#2345] [Question]
So I have a test fixture to spin up a simple in memory webhost which acts as a mock api for integration tests to hit (lets not get into the whys etc).

This works fine when I run each test individually, or all tests within a class, however the problem arises when I have multiple test classes using the same fixture as only one fixture can be active on a given port, so the first class to run steals the port, then some of the other classes pass as they use the same host, but when the initial class finishes it tears down the host and then all other tests fail.

My current thought is to have a static counter and basically increment the ports passing the fixture into each related test class and just using that port, but I wasnt sure if anyone knew of a better way? ideally I would love to just spin it up before all tests, and tear it down after they are all run but to my knowledge you cant do that :(

**Comment by bradwilson** [comment]
I believe you should put all the tests into a test collection, and use `ICollectionFixture` instead of `IClassFixture` on the collection class.

Docs: https://xunit.net/docs/shared-context#collection-fixture
